extra_geom/detectors.py

```diff
@@ -1342,3 +1342,119 @@ def __init__(self, modules, filename='No file'):
         warnings.warn(
             "DSSC_Geometry has been renamed to DSSC_1MGeometry.", stacklevel=2
         )
+
+
+class Jungfrau_Geometry(DetectorGeometryBase):
+    """Detector layout for flexible Jungfrau arrangements based on the JF-500K
+       module, which is an independent detector unit of 2 x 4 ASIC tiles.
+
+       The ss/fs assignment is the other way around as for AGIPD:
+       y = ss
+       x = fs
+       shape = (ss, fs) = (y, x)
+    """
+    pixel_size = 7.5e-5   # 7.5e-5 metres = 75 micrometer = 0.075 mm
+    frag_ss_pixels = 256  # pixels along slow scan axis within tile
+    frag_fs_pixels = 256  # pixels along fast scan axis within tile
+
+    def __init__(self, modules, filename='No file'):
+        super().__init__(modules, filename)
+        self.expected_data_shape = (len(modules), 512, 1024)
+
+    @classmethod
+    def from_module_positions(cls,offsets=((0,0),), orientations=None,
+                              asic_gap=2, unit=pixel_size):
+        """Generate a Jungfrau geometry object from module positions
+
+           Parameters
+           ----------
+
+           offsets - iterable of length n_modules containing a pixel coordinate
+                     tuple (x,y) for each offset to the global origin
```

Comment:
I think we should specify which corner of the module the offset is to. I think the code currently uses it as the position of the lower beam-right corner?

My preference is to specify the position of the (0, 0) pixel, but that does mean that which corner you're measuring to depends on the orientation, which I appreciate is not necessarily the most convenient way of working. So I'm happy to leave the implementation as is, but I would like it to be clarified in the docstring.

I see this is in an internal comment in the loop already, but that's not visible without inspecting the source code. And I prefer to avoid unqualified 'left' and 'right', as I mentioned, because people look at it in both orientations. I hope that terms like 'beam-left' are a clear shorthand for 'left when looking along the beam'.

Reply:
I shortened the internal comment and put most of its meaning into the docstring, rephrasing a bit (e.g. using beam-left).